source/plugin/basics/text/pagination.rst

```diff
@@ -0,0 +1,101 @@
+======================
+The Pagination Service
+======================
+
+.. tip::
+
+	For a basic understanding of services, make sure you read :doc:`../../advanced/services` first.
+
+The ``PaginationService`` acts as a way to split up content into discrete pages.
+The service provides a ``PaginationBuilder`` with which you can specify attributes such as title, contents, header, and padding.
+
+Working With the ``PaginationBuilder``
+======================================
+
+First obtain an instance of the ``PaginationService``, and create a new ``PaginationBuilder``:
+
+.. code-block:: java
+
+	import org.spongepowered.api.service.pagination.PaginationBuilder;
+	import org.spongepowered.api.service.pagination.PaginationService;
+
+	PaginationService paginationService = plugin.getGame().getServiceManager().provide(PaginationService.class).get();
+	PaginationBuilder builder = paginationService.newBuilder();
+
+There are two different ways to specify the contents of paginated list:
+
+* With an ``Iterable<Text>``
+
+ .. code-block:: java
+
+	import java.util.ArrayList;
+	import java.util.List;
+	import org.spongepowered.api.text.Text;
+	import org.spongepowered.api.text.Texts;
+
+	List<Text> contents = new ArrayList<>();
+	contents.add(Texts.of("Item 1"));
+	contents.add(Texts.of("Item 2"));
+	contents.add(Texts.of("Item 3"));
+
+	builder.contents(contents);
+
+ .. note::
+
+	If the ``Iterable`` is a ``List``, then bidirectional navigation is supported. Otherwise, only forwards navigation is supported.
+
+* With an array of ``Text``\ s
+
+ .. code-block:: java
+
+	import org.spongepowered.api.text.Texts;
+
+	builder.contents(Texts.of("Item 1"), Texts.of("Item 2"), Texts.of("Item 3"));
+
+You can also specify various other components of a paginated list, such as a title, header, footer, and padding. The diagram below shows which component is displayed
+in each part of the paginated list. In the following diagram, the padding string is shown as the letter `p`.
+
+.. code-block:: java
+
+	pppppppppppppppppppppppp Title pppppppppppppppppppppppp
+	Header
+	Item 1
+	Item 2
+	Item 3
+	...
+	ppppppppppppppppppppppp < 2/3 > ppppppppppppppppppppppp
+	Footer
+
+To achieve the preceding output, we might use the following builder pattern:
+
+.. code-block:: java
+
+	import org.spongepowered.api.text.Texts;
+
+	builder.title(Texts.of("Title"))
+		.contents(Texts.of("Item 1"), Texts.of("Item 2"), Texts.of("Item 3"))
+		.header(Texts.of("Header"))
```

Comment:
might want to make it clear that everything but contents is optional? but title at least is strongly recommended